src/clj/rems/email/core.clj

```diff
@@ -20,12 +22,23 @@
     (template/event-to-emails (rems.application.model/enrich-event event users/get-user (constantly nil))
                               (applications/get-unrestricted-application app-id))))
 
-(defn generate-emails! [new-events]
+(defn- enqueue-email! [email]
+  (outbox/put! {:outbox/type :email
+                :outbox/email email
+                :outbox/deadline (-> (time/now) (.plus ^Period (:email-retry-period env)))}))
+
+(defn generate-event-emails! [new-events]
   (doseq [event new-events
           email (event-to-emails event)]
-    (outbox/put! {:outbox/type :email
-                  :outbox/email email
-                  :outbox/deadline (-> (time/now) (.plus ^Period (:email-retry-period env)))})))
+    (enqueue-email! email)))
+
+;; TODO: scheduler or API+cron
```

Comment:
Hmm, I think there's potentially a fundamental problem here: I'd imagine that we wouldn't want to send out a reminder email for an application filed on the previous day, but having a weekly cron job or similar doesn't distinguish between actually old and recently filed applications.

Having a reminder email of all open applications at a particular moment also makes it difficult to see, which of the applications have been open long enough so that they warrant extra attention.

Also, if we had a filter here that would filter out recent enough applications and only leave applications older than, say, one week, depending on when the scheduler or cron job is run relatively to the time when the application was submitted, it could be take one week and six days before a reminder email for that application is sent.

I think the best solution would be to send out reminder emails on-application basis, one email per each application that is older than the specified time limit.

Reply:
Never mind, Leena enlightened me that this weekly reminder email for all open applications is exactly how it has been specified.

Resolving the conversation.